Tasks/XamariniOSV2/xamarinios.ts

```diff
@@ -20,14 +21,19 @@ function expandSolutionWildcardPatterns(solutionPattern: string): string {
 
         return result;
     } else {
-        throw tl.loc('SolutionDoesNotExist', solutionPattern);
+        throw new Error (tl.loc('SolutionDoesNotExist', solutionPattern));
     }
 }
 
 async function run() {
     try {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
+        // Check platform is macOS since demands are not evaluated on Hosted pools
+        if (!/^darwin/.test(os.platform())) {
```

Comment:
Why use a regex?  The list of possible platforms [is documented](https://nodejs.org/api/os.html#os_os_platform), so you can just check `=== 'darwin'`

Reply:
Changing to !== 'darwin'